### PR TITLE
allow fmpq -> fmpz

### PR DESCRIPTION
**Pull request by tthsqe12**
this time with no extra garbage

### CODE REVIEW

File: src/flint/fmpq.jl [hunk 1]
```diff
@@ -883,6 +883,15 @@ end
 
 (a::FlintRationalField)(b::fmpq) = b
 
+function (a::FlintIntegerRing)(b::fmpq)
+   z = fmpz()
+   ccall((:fmpq_denominator, libflint), Nothing, (Ref{fmpz}, Ref{fmpq}), z, b)
+   isone(denominator(b)) || error("Denominator must be 1")
```
Comment (thofma) [inline src/flint/fmpq.jl:889]: Is there a flint function to do this allocation free?

Comment (tthsqe12) [reply src/flint/fmpq.jl:889]: oops. I royally messed that line up. I was trying to make zero garbage.

File: src/flint/fmpq.jl [hunk 2]
```diff
@@ -883,6 +883,15 @@ end
 
 (a::FlintRationalField)(b::fmpq) = b
 
+function (a::FlintIntegerRing)(b::fmpq)
+   z = fmpz()
+   ccall((:fmpq_denominator, libflint), Nothing, (Ref{fmpz}, Ref{fmpq}), z, b)
+   isone(z) || error("Denominator must be 1")
```
Comment (albinahlback) [inline src/flint/fmpq.jl:889]: Is it possible to do this without having to copy the denominator of `b`?

Comment (albinahlback) [reply src/flint/fmpq.jl:889]: oh, Tommy already asked this

Comment (tthsqe12) [reply src/flint/fmpq.jl:889]: No, not officially. It's ok as it is easy to copy 1, and it is also already an improvement over the hecke version, which creates two extra garbage fmpz's.

**Pull request merged**